### PR TITLE
incubator-kie-issues#471: Add apache headers license into files

### PR DESCRIPTION
**Pull request by jomarko**
**JIRA:**
Closes: https://github.com/apache/incubator-kie-issues/issues/471

Please notice I am not adding headers to all `.adoc` files, because on apache pages is written it is not needed for code snippet that is included into some parent, that has header https://www.apache.org/legal/src-headers.html#faq-exceptions

This PR also replaces github **kiegroup** references to github **apache** references

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributions doc](https://github.com/apache/incubator-kie-kogito-docs/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by ricardozanini** [comment]
Can you please rebase? 🙏

**Comment by jomarko** [comment]
@ricardozanini rebased

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/developing-workflows.adoc [hunk 1]
```diff
@@ -138,7 +138,11 @@ For example using VSCode, there are the commands needed:
 
 [source,bash,subs="attributes+"]
 ----
+<<<<<<< HEAD
 curl -S https://raw.githubusercontent.com/apache/incubator-kie-kogito-serverless-operator/{operator_version}/test/testdata/sonataflow.org_v1alpha08_sonataflow_devmode.yaml > workflow_devmode.yaml
+=======
+curl -S https://raw.githubusercontent.com/apache/incubator-kie-kogito-serverless-operator/{operator_version}/config/samples/sw.kogito_v1alpha08_sonataflow_devmode.yaml > workflow_devmode.yaml
+>>>>>>> 744e7dc (kiegroup -> apache)
 code workflow_devmode.yaml
```
Comment (porcelli) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/developing-workflows.adoc:146]: I think this conflict hasn't been solved and has been left behind

**Review by porcelli — APPROVED** [review]
lgtm

**Review by MarianMacik — COMMENTED** [review]
LGTM.

**Pull request merged**